upstairs/src/upstairs.rs

```diff
@@ -455,18 +495,30 @@ impl Upstairs {
     pub(crate) async fn apply(&mut self, action: UpstairsAction) {
         match action {
             UpstairsAction::Downstairs(d) => {
+                self.counters.action_downstairs += 1;
+                cdt::up__action_downstairs!(|| (self
+                    .counters
+                    .action_downstairs));
                 self.apply_downstairs_action(d).await
             }
             UpstairsAction::Guest(b) => {
+                self.counters.action_guest += 1;
+                cdt::up__action_guest!(|| (self.counters.action_guest));
                 self.defer_guest_request(b).await;
             }
             UpstairsAction::DeferredBlockReq(req) => {
+                self.counters.action_deferred += 1;
+                cdt::up__action_deferred!(|| (self.counters.action_deferred));
                 self.apply_guest_request(req).await;
             }
             UpstairsAction::DeferredMessage(m) => {
```

Comment:
`UpstairsAction::DeferredMessage` was probably added after you wrote this PR, can you add it to the counters + probes?

Reply:
I updated the counters, scripts, and example to have the new `UpstairsAction::DeferredMessage` type.